packages/packages/store/src/index.ts

```diff
@@ -0,0 +1,110 @@
+import {
+	createSlice,
+	ReducersMapObject,
+	configureStore,
+	combineReducers,
+	ThunkMiddleware,
+	Store,
+	Slice,
+	AnyAction,
+	CreateSliceOptions,
+} from '@reduxjs/toolkit';
+
+export type {
+	Slice,
+	CreateSliceOptions,
+	PayloadAction,
+	Store,
+	Dispatch,
+	AnyAction,
+	Action,
+	MiddlewareAPI,
+} from '@reduxjs/toolkit';
+
+export { createSlice } from '@reduxjs/toolkit';
+
+export { useSelector, useDispatch, Provider as StoreProvider } from 'react-redux';
+
+// Usage: SliceState<typeof slice>
+export type SliceState<S extends Slice> = {
+	[ key in S['name'] ]: ReturnType<S['getInitialState']>;
+}
+
+interface SlicesMap {
+	[key: Slice['name']]: Slice;
+}
+
+class StoreService {
+	private instance: Store | null = null;
+	private slices: SlicesMap = {};
+	private pendingActions: AnyAction[] = [];
+	private middlewares = new Set<ThunkMiddleware>();
+
+	private getReducers() {
+		const reducers = Object.entries( this.slices ).reduce( ( reducersData: ReducersMapObject, [ name, slice ] ) => {
+			reducersData[ name ] = slice.reducer;
+
+			return reducersData;
+		}, {} );
+
+		return combineReducers( reducers );
+	}
+
+	// Should be casted into createSlice because this function detects the correct type only when the slice object is passed directly to it.
+	registerSlice( sliceConfig: CreateSliceOptions ) {
+		const slice = createSlice( sliceConfig );
+
+		if ( this.slices[ slice.name ] ) {
+			return this.slices[ slice.name ];
+		}
```

Comment:
So you cannot override a slice? why not? pro? or at least throw an error if the user try to override if we think we do not want to do so.

Reply:
Changed it to throw an error + added a test.